haystack/utils.py

```diff
@@ -79,3 +81,80 @@ def convert_labels_to_squad(labels_file: str):
 
     with open("labels_in_squad_format.json", "w+") as outfile:
         json.dump(labels_in_squad_format, outfile)
+
+def find_answer_start(answer: str, text: str):
+    """
+    get index of beginning of answer in text
+    :param text: string where 'answer' is to be searched in
+    :param answer: substring to be searched in 'text'
+    :return: list of indices of occurrences of answer
+    """
+    # escape all (
+    answer = r"\b{}\b".format(re.escape(answer))
+    return [m.start() for m in re.finditer(answer, text)]
+
+def convert_df_to_squad(dataframe):
+    """
+    convert pandas data-frame to squad json
+    :param dataframe: Pandas dataframe where each row represents one question-context sample with other relevant info
+                    columns in the dataframe:
+                        question: question string: str
+                        answers: list of answers: [str]
+                        is_impossible: is answer absent: Optional[bool]
+                        context: context passage: str
+                        passage_id: unique identification number for a context passage: int
+                        title: title of the context passage: str
+    :return: json in Squad Format
+    """
+    squad_data = []
+
+    # Group rows with same context 'title'
+    for title, all_paragraphs in dataframe.groupby(['title']):
+        paras_in_title = []
+
+        # Group rows with same context passages identified with 'passage_id'
+        for para_id, para in all_paragraphs.groupby(['passage_id']):
+            ques_ans_in_paragraph = []
+
+            # create Squad 'qas'
+            for questions, answers in zip(para['question'], para['answers']):
+
+                # group Squad 'answers' within a list of answers
+                ans_ = [answer(text=ans, answer_start=index) for ans in answers
+                        for index in find_answer_start(ans, para.context.iloc[0])]
+                # qas
+                ques_ans_in_paragraph.append(qas(question=questions, answers=ans_))
+
+            # Group all passages with same title within a list of passages to create Squad 'passages'
+            paras_in_title.append(paragraphs(qas=ques_ans_in_paragraph,
+                                             context=para.context.iloc[0],
+                                             passage_id=para_id))
+
+        # squad 'data': list of articles grouped by title
+        squad_data.append(data(title=title, paragraphs=paras_in_title))
+    return SquadSchema(data=squad_data).json()
+
+def convert_dpr_to_squad(json_data, **kwargs):
```

Comment:
Let's change this function so that it takes "input_file" and "output_file" as args and writes the results to the output file